Objects/obmalloc.c

```diff
@@ -1459,37 +1459,77 @@ _PyMem_DebugRawFree(void *ctx, void *p)
 static void *
 _PyMem_DebugRawRealloc(void *ctx, void *p, size_t nbytes)
 {
+#ifdef __OpenBSD__
+    /* bpo-31626: On OpenBSD, it's illegal to erase old bytes in the old
+       memory block on realloc() success: the program is killed by a fatal
+       trap signal. So erase bytes before, but keep a copy to restore erased
+       bytes if realloc() fails. */
+#   define REALLOC_WRITE_BEFORE
```

Comment:
Maybe it would be simpler to modify _PyMem_DebugRawRealloc() to behave "correctly" on all platforms. What do you think?

I don't think that the "copy" memory allocation is a real blocker issue. It's a debug hook, I consider that the memory usage is not a big deal here.